### PR TITLE
Refactor account and transaction types

### PR DESCRIPTION
**Pull request by arhtudormorar**
- expose getTransactions methods

### CODE REVIEW

File: src/apiCalls/getScamAddressData.ts [hunk 1]
```diff
@@ -7,13 +7,13 @@ export async function getScamAddressData(addressToVerify: string) {
...
-    `/accounts/${addressToVerify}`,
-    {
-      baseURL: apiAddress,
-      timeout: Number(apiTimeout)
-    }
-  );
+  const { data } = await axios.get<{
+    scamInfo?: ScamInfoType;
+    code?: string;
+  }>(`/accounts/${addressToVerify}`, {
```
Comment (CiprianDraghici) [inline src/apiCalls/getScamAddressData.ts:13]: /accounts can be extracted into endpoints.ts file

File: src/UI/TransactionsTable/TransactionIcon.tsx [hunk 2]
```diff
@@ -9,11 +9,11 @@ import globalStyles from 'assets/sass/main.scss';
 import { getOperationsMessages } from 'components/TransactionsInterpreter/helpers/getOperationsMessages';
 import { getReceiptMessage } from 'components/TransactionsInterpreter/helpers/getReceiptMessage';
 import getScResultsMessages from 'components/TransactionsInterpreter/helpers/getScResultsMessages';
-import { UITransactionType } from 'components/TransactionsInterpreter/helpers/types';
-import { TransactionServerStatusesEnum } from '../../types';
+import { TransactionServerStatusesEnum } from 'types/enums';
+import { TransactionType } from 'types/server-transations';
```
Comment (StanislavSava) [inline src/UI/TransactionsTable/TransactionIcon.tsx:13]: I think we should use camelCase for all files

File: src/UI/TransactionsTable/TransactionMethod.tsx [hunk 3]
```diff
@@ -2,13 +2,13 @@ import React from 'react';
 import classNames from 'classnames';
 import globalStyles from 'assets/sass/main.scss';
 import { getTransactionMethod } from 'components/TransactionsInterpreter/helpers/getTransactionMethod';
-import { UITransactionType } from 'components/TransactionsInterpreter/helpers/types';
+import { TransactionType } from 'types/server-transations';
```
Comment (StanislavSava) [inline src/UI/TransactionsTable/TransactionMethod.tsx:5]: there's a typo in the file name

File: src/components/TransactionsInterpreter/tests/TransactionsInterpreter.test.ts [hunk 4]
```diff
@@ -1,9 +1,8 @@
 import { mainnetEgldLabel } from 'constants/network';
+import { TransactionType, TxActionsEnum } from 'types/server-transations';
```
Comment (StanislavSava) [inline src/components/TransactionsInterpreter/tests/TransactionsInterpreter.test.ts:2]: Shouldn't we name this "ServerTransactionType"?

File: src/types/server-transations.ts [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+}
+
+export enum TxActionsEnum {
+  // esdtNft category
+  transfer = 'transfer',
+  // stake category
+  delegate = 'delegate',
+  stake = 'stake',
+  unDelegate = 'unDelegate',
+  stakeClaimRewards = 'claimRewards',
```
Comment (StanislavSava) [inline src/types/server-transations.ts:32]: I think we should split this into multiple enums and make a union, this enum is too big

File: src/utils/account/refreshAccount.tsx [hunk 6]
```diff
@@ -12,10 +12,9 @@ const setNewAccount = async () => {
...
       if (account != null) {
         const accountData = {
-          balance: account.balance,
+          ...account,
           address,
-          nonce: getLatestNonce(account),
-          username: account.username
+          nonce: getLatestNonce(account)
         };
         store.dispatch(setAccount(accountData));
```
Comment (StanislavSava) [inline src/utils/account/refreshAccount.tsx:19]: maybe we should create a redux action "updateAccountData" and just dispatch the values that need to be updated and to the merging inside redux

File: src/components/TransactionsInterpreter/tests/extended-transaction-mock.ts [hunk 7]
```diff
@@ -1,13 +1,13 @@
...
+  VisibleTransactionOperationType,
+  TransactionOperationActionType,
+  TransactionType
+} from 'types/server-transations';
+import { baseTransactionMock } from '../helpers/tests/base-transaction-mock';
 
-export const transactionMock: UITransactionType = {
+export const transactionMock: TransactionType = {
   ...baseTransactionMock,
   // type: TransferTypeEnum.SmartContractResult,
```
Comment (CiprianDraghici) [inline src/components/TransactionsInterpreter/tests/extended-transaction-mock.ts:12]: I know this is not added by you, but I appreciate it if you will remove this commented line

**Review by CiprianDraghici — APPROVED** [review]

File: src/utils/account/refreshAccount.tsx (hunk 6 above)
Comment (arhtudormorar) [reply src/utils/account/refreshAccount.tsx:18]: account comes from API. It's OK to only update the nonce

**Review by StanislavSava — APPROVED** [review]

**Pull request merged**